discord/client.py

```diff
@@ -442,6 +442,30 @@ def _reconnect_ws(self):
     # login state management
 
     @asyncio.coroutine
+    def login_with_token(self, token):
+        self.token = token
+        if self.token is None:
+            raise LoginFailure('No Token Specified.')
+            yield from self.close()
+        else:
+            self.headers['authorization'] = token
+            self._is_logged_in.set()
+            try:
+                self.loop.run_until_complete(self.connect())
```

Comment:
It does not make much sense for this to call connect; it looks like you are duplicating the functionality of `run` and `start`.  IMO it would make a lot more sense to keep `login` as only doing a "log in" while use `run` / `start` to handle calling `connect` and `KeyboardInterrupts` etc.